drivers/gpio/gpio_pcf8574.c

```diff
@@ -0,0 +1,389 @@
+static int pcf8574_port_set_masked_raw(const struct device *dev, gpio_port_pins_t mask,
+				       gpio_port_value_t value)
+{
+	return pcf8574_port_set_raw(dev, (uint8_t)mask, (uint8_t)value, 0);
+}
+
+/**
+ * @brief Sets some output pins of the pcf8574
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param pins The pin(s) which will be set in a range from 0 to 7
+ *
+ * @retval 0 If successful.
+ * @retval Negative value for error.
+ */
+static int pcf8574_port_set_bits_raw(const struct device *dev, gpio_port_pins_t pins)
+{
+	return pcf8574_port_set_raw(dev, (uint8_t)pins, (uint8_t)pins, 0);
+}
+
+/**
+ * @brief clear some bits
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param pins Pins which will be cleared
+ *
+ * @retval 0 If successful.
+ * @retval Negative value for error.
+ */
+static int pcf8574_port_clear_bits_raw(const struct device *dev, gpio_port_pins_t pins)
+{
+	return pcf8574_port_set_raw(dev, (uint8_t)pins, 0, 0);
+}
+
+/**
+ * @brief Toggle some bits
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param pins Pins which will be toggled
+ *
+ * @retval 0 If successful.
+ * @retval Negative value for error.
+ */
+static int pcf8574_port_toggle_bits(const struct device *dev, gpio_port_pins_t pins)
+{
+	return pcf8574_port_set_raw(dev, 0, 0, (uint8_t)pins);
+}
+
+/* Each pin gives an interrupt at pcf8574.
+   In this function is checked, whether the configuration is right*/
+static int pcf8574_pin_interrupt_configure(const struct device *dev, gpio_pin_t pin,
+					   enum gpio_int_mode mode, enum gpio_int_trig trig)
+{
+	const struct pcf8574_drv_cfg *drv_cfg = dev->config;
+	struct pcf8574_drv_data *drv_data = dev->data;
+
+	int rc = 0;
+
+	if (!drv_cfg->gpio_int.port) {
+		return -ENOTSUP;
+	}
+
+	/* This device supports only edge-triggered interrupts. */
+	if (mode == GPIO_INT_MODE_LEVEL) {
+		return -ENOTSUP;
+	}
+
+	return rc;
+}
+
+/** Register the callback in the callback list*/
+static int pcf8574_manage_callback(const struct device *dev, struct gpio_callback *callback,
+				   bool set)
+{
+	struct pcf8574_drv_data *drv_data = dev->data;
+
+	return gpio_manage_callback(&drv_data->callbacks, callback, set);
+}
+
+/** Initialize the pcf8574*/
+static int pcf8574_init(const struct device *dev)
+{
+	const struct pcf8574_drv_cfg *drv_cfg = dev->config;
+	struct pcf8574_drv_data *drv_data = dev->data;
+	int rc;
+
+	if (!device_is_ready(drv_cfg->i2c.bus)) {
+		LOG_ERR("%s is not ready", drv_cfg->i2c.bus->name);
+		return -ENODEV;
+	}
+
+	/* Initialize lock */
+	k_sem_init(&drv_data->lock, 1, 1);
+
+	/* If the INT line is available, configure the callback for it. */
+	if (drv_cfg->gpio_int.port) {
+		if (!device_is_ready(drv_cfg->gpio_int.port)) {
+			LOG_ERR("Port is not ready");
+			return -ENODEV;
+		}
+
+		rc = gpio_pin_configure(drv_cfg->gpio_int.port, drv_cfg->gpio_int.pin,
+					drv_cfg->gpio_int.dt_flags | GPIO_INPUT);
```

Comment:
Use the `gpio_dt_spec` helper functions.

```suggestion
		rc = gpio_pin_configure_dt(&drv_cfg->gpio_int, GPIO_INPUT);
```